src/plugins/i18n/server/i18n/i18n.js

```diff
@@ -0,0 +1,125 @@
+  });
+};
+
+const getRegisteredLanguageTranslations = function (language) {
+  const translationStorePath = getTranslationStoragePath();
+  const translationFileName = language + '.' + TRANSLATION_FILE_EXTENSION;
+  const translationFile = translationStorePath + PATH_SEPARATOR + translationFileName;
+
+  return readFile(translationFile, 'utf8').then(function (translationStr) {
+    let translationJson = [];
+    try {
+      translationJson = JSON.parse(translationStr);
+    } catch (err) {
+      throw new Error('Bad ' + language + ' translation strings. Reason: ' + err);
+    }
+    return translationJson;
+  }).catch(function (e) {
+    throw new Error('Failed to read translation file. Reason: ' + e);
+  });
+};
+
+function storeTranslations(absolutePluginTranslationFilePath, translationStorePath) {
+  const translationFileName = getFileName(absolutePluginTranslationFilePath);
+  const translationJson = require(absolutePluginTranslationFilePath);
+  const fileToWrite = translationStorePath + PATH_SEPARATOR + translationFileName;
+  return getTranslationJsonToWrite(fileToWrite, translationJson).then((jsonToWrite) => {
+    return writeFile(fileToWrite, JSON.stringify(jsonToWrite));
+  });
+}
+
+function getTranslationJsonToWrite(translationFullFileName, translationToAddJson) {
+  return stat(translationFullFileName).then((stats) => {
+    let jsonToWrite = [];
+    const currentTranslationJson = require(translationFullFileName);
+    jsonToWrite = currentTranslationJson;
+    for (let key in translationToAddJson) {
+      if (translationToAddJson.hasOwnProperty(key)) {
+        const attrName = key;
+        const attrValue = translationToAddJson[key];
+        jsonToWrite[attrName] = attrValue;
+      }
+    }
+    return jsonToWrite;
+  }).catch(function (e) {
+    return translationToAddJson;
+  });
+}
+
+function createTranslationDirectory(translationStorePath) {
+  return stat(translationStorePath).then((stats) => {
+  }).catch(function (e) {
+    return mkdirpAsync(translationStorePath);
+  });
+}
+
+function getTranslationDetailsFromDirectory(dir, translationFiles, languageList) {
+  return readdir(dir).then((dirListing) => {
+    return Promise.map(dirListing, (listing) => {
+      const fullPath = path.join(dir, listing);
+      return stat(fullPath).then((stats) => {
+        if (!stats.isDirectory()) {
+          getTranslationDetailsFromFile(fullPath, translationFiles, languageList);
+        }
+      });
+    });
+  });
+}
+
+function getTranslationDetailsFromFile(fullFileName, translationFiles, languageList) {
+  const fileName = getFileName(fullFileName);
+  const fileExt = fileName.split('.').pop();
+
+  if (fileName === fileExt) return;
+  if (fileExt !== TRANSLATION_FILE_EXTENSION) return;
+  translationFiles.push(fullFileName);
+  const lang  = fileName.substr(0, fileName.lastIndexOf('.'));
+  if (languageList.indexOf(lang) === -1) {
+    languageList.push(lang);
```

Comment:
so this shouldn't just be a lang but actually a full locale code, such as `zh-Hant-TW` or `ast-ES` right? it says language .